Lambda with an anonymous parameter type gets mixed parameter typing (CS0748)

### Summary

When one lambda parameter has an anonymous type, ILSpy correctly omits that parameter's
unnameable type but leaves sibling parameter types explicit. C# requires lambda parameter
types to be either all explicit or all implicit, so the output fails with CS0748.

### Input code

Compiled in Release targeting .NET 10.0:

```csharp
using System;

public static class C
{
    private static TValue GetOrCreate<TKey, TValue>(
        TKey key,
        Func<TKey, int, TValue> factory) => factory(key, 1);

    public static int Read()
    {
        var key = new { Value = 1 };
        return GetOrCreate(key, (item, context) => item.Value + context);
    }
}
```

### Erroneous output

```csharp
public static int Read()
{
    return GetOrCreate(new
    {
        Value = 1
    }, (item, int context) => item.Value + context);
}
```

Recompiling this output fails:

```text
error CS0748: Inconsistent lambda parameter usage; parameter types must be all explicit or all implicit
```

### Expected output

Because the anonymous type cannot be named, every parameter should remain implicit:

```csharp
(item, context) => item.Value + context
```

### Details

* Product in use: ICSharpCode.Decompiler
* Version in use: master `7554c1b0` (`11.0.0.9223`)
* The repro was verified in an isolated compile, decompile, recompile round trip.
* `ExpressionBuilder.MakeParameters` removes the type only from parameters whose type contains an anonymous type, leaving the remaining parameter declarations explicitly typed.
* I am happy to contribute a fix.

--- a/ICSharpCode.Decompiler.Tests/TestCases/Pretty/AnonymousTypes.cs
+++ b/ICSharpCode.Decompiler.Tests/TestCases/Pretty/AnonymousTypes.cs
@@ -174,5 +174,18 @@ namespace ICSharpCode.Decompiler.Tests.TestCases.Pretty
 		{
 			return null;
 		}
+#if ROSLYN
+		private static TValue GetOrCreate<TKey, TValue>(TKey key, Func<TKey, int, TValue> factory)
+		{
+			return factory(key, 1);
+		}
+
+		private static int MixedLambdaParameters()
+		{
+			return GetOrCreate(new {
+				Value = 1
+			}, (item, context) => item.Value + context);
+		}
+#endif
 	}
 }

--- a/ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs
+++ b/ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs
@@ -2775,6 +2775,8 @@ namespace ICSharpCode.Decompiler.CSharp
 		IEnumerable<ParameterDeclaration> MakeParameters(IReadOnlyList<IParameter> parameters, ILFunction function)
 		{
 			var variables = function.Variables.Where(v => v.Kind == VariableKind.Parameter).ToDictionary(v => v.Index!.Value);
+			var result = new List<ParameterDeclaration>(parameters.Count);
+			bool anyAnonymousType = false;
 			int i = 0;
 			foreach (var parameter in parameters)
 			{
@@ -2790,10 +2792,21 @@ namespace ICSharpCode.Decompiler.CSharp
 					pd.Name = "P_" + i;
 				}
 				if (settings.AnonymousTypes && parameter.Type.ContainsAnonymousType())
-					pd.Type = null;
-				yield return pd;
+					anyAnonymousType = true;
+				result.Add(pd);
 				i++;
 			}
+			// An anonymous type cannot be named, so a lambda with such a parameter must be implicitly typed.
+			// C# also requires all lambda parameters to use the same form (CS0748). Drop every type when the
+			// remaining parameter syntax permits it; otherwise keep the converted declarations as a
+			// best-effort fallback for an unrepresentable signature.
+			if (anyAnonymousType && result.All(p => p.ParameterModifier == ReferenceKind.None
+				&& !p.IsParams && !p.IsScopedRef && p.Attributes.Count == 0 && p.DefaultExpression is null))
+			{
+				foreach (var pd in result)
+					pd.Type = null;
+			}
+			return result;
 		}
 
 		protected internal override TranslatedExpression VisitBlockContainer(BlockContainer container, TranslationContext context)

--- a/ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs
+++ b/ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs
@@ -2791,17 +2791,19 @@ namespace ICSharpCode.Decompiler.CSharp
 					// needs to be consistent with logic in ILReader.CreateILVariable
 					pd.Name = "P_" + i;
 				}
+
 				if (settings.AnonymousTypes && parameter.Type.ContainsAnonymousType())
 					anyAnonymousType = true;
+
 				result.Add(pd);
 				i++;
 			}
+
 			// An anonymous type cannot be named, so a lambda with such a parameter must be implicitly typed.
 			// C# also requires all lambda parameters to use the same form (CS0748). Drop every type when the
 			// remaining parameter syntax permits it; otherwise keep the converted declarations as a
 			// best-effort fallback for an unrepresentable signature.
-			if (anyAnonymousType && result.All(p => p.ParameterModifier == ReferenceKind.None
-				&& !p.IsParams && !p.IsScopedRef && p.Attributes.Count == 0 && p.DefaultExpression is null))
+			if (anyAnonymousType && result.All(p => p is { ParameterModifier: ReferenceKind.None, IsParams: false, IsScopedRef: false, Attributes.Count: 0, DefaultExpression: null }))
 			{
 				foreach (var pd in result)
 					pd.Type = null;

